doc/development/rfc/version_numbering.md

```diff
@@ -0,0 +1,146 @@
+
+The increasing versions was always following the same ideas as in
+the Semantic Versioning, i.e., _major_, _minor_, and _patch_
+with major for breaking changes, minor for backwards compatible
+features, and patch for fixes.
+
+For the versions 7.0 till 8.2, there was a distinction between even and odd minor versions.
+Even minor versions were released, while odd minor version marked development version.
+This odd versions never had a patch number assigned and were never released.
+
+## Motivation
+
+Explaining the version numbering should be as easy as possible.
+Ideally, it would not need any explanation at all.
+When all version numbers refer to releases, previews (alphas, betas, daily builds),
+always need additional indication of the version being a preview.
+However, using odd version to mark the development versions requires
+still additional explanation, for example, download for 8.1 on the website
+said _preview_ anyway because the odd number does not indicate the
+development version.
+Hence, odd numbers for development versions do not bring any advantage.
+
+The branching and release procedure with even minor numbers for releases
+and odd numbers for development versions require that when new branch for
+a minor version is created, the branch needs to change all the mentions
+of the minor version number to the next even number and, at the same time,
+the _main_ branch needs to change to the next odd number.
+Without the even and odd distinction, the branch keeps the version from
+the main branch while the main branch advanced to the next upcoming version
+resulting only in one operation which is the same whether the current version
+is even or odd.
+
+The Semantic Versioning uses labels after the version number to indicate development
+versions, so using that system or a similar one should be sufficient to mark the
+development versions.
+
+## Version Numbering Specification
+
+### Major
+
+In accordance with Semantic Versioning, major release must happen with
+any backwards incompatible change in the API which includes
+both interface and behavior changes (API also defines what happens,
+not just names and signatures).
+
+Additionally, major feature additions which would require only minor release in terms of API stability,
+are strongly suggested to trigger major release as well.
+This in turn helps to address the issue of outdated tutorials and small, but breaking, changes.
+
+Some major features such as changes in the GUI are fully backwards compatible
+(GUI API is not a public API in versions 7 and 8), but are heavily influencing tutorials and
+other teaching materials. In that sense, behavior linked to a particular interface is changed,
+so a major version is required.
+
+There is always a list of many small changes which are not backwards compatible,
+for example, a cleanup of depreciated functions. None of these changes alone does
+not seem worth a major release, but because it would require one, it is never done.
```

Comment:
```suggestion
for example, a cleanup of depreciated functions. None of these changes alone 
seems worth a major release, but because it would require one, it is never done.
```